implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -403,22 +399,20 @@ protected DistributionStatisticConfig defaultHistogramConfig() {
                 .merge(DistributionStatisticConfig.DEFAULT);
     }
 
+    /**
+     * @deprecated queue size is no longer available
+     */
+    @Deprecated
     public int queueSize() {
-        try {
-            return (Integer) processor.getClass().getMethod("size").invoke(processor);
-        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
-            // should never happen
-            return 0;
-        }
+        return -1;
```

Comment:
We still need to consider what is the best OverflowStrategy for us, but assuming we choose one with buffering, is it possible to check that the `DirectProcessor` has any items buffered or not? I see a `getBufferSize` method I can call on the `DirectProcessor`, but this is actually the capacity, not the current size of items queued.

Reply:
On Reactor side we really need to do some heavy reworking of processors and the associated APIs, but that will be quite an heavy breaking change.

`DirectProcessor` requires at least one `Subscriber` with enough pending `request` any time some data is fed to it (no backpressure). It falls flat if that is not the case. As a consequence, it is also assumed that buffering in the sink makes no sense... Its `getBufferSize()` returning `Integer.MAX_VALUE` is actually interpreted as "no buffering". And its `sink(OverflowStrategy)` actually ignores the strategy (see [this](https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/FluxProcessor.java#L229)). The forced `IGNORE` is not terribly relevant, since it will still attempt to call `DirectProcessor#onNext`, which simply does nothing if there is 0 subscriber (see [here](https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/DirectProcessor.java#L141)).

TL;DR: *There is no way to change the overflow strategy of the `DirectProcessor`'s sink, it will always drop if no subscriber.

Reply:
Thanks for the explanation. I think that works for our purposes. We do some buffering ourselves in our `BufferingFlux`. If more metrics are produced than can be buffered in that and sent, I think dropping makes more sense than potentially OOMing the application. It would be better if we knew we were overflowing somehow though, so we could expose that as a metric or at least a log message. Is there any way we could achieve that?

Reply:
not with the sink. the operators have consumers (eg. `.onBackpressureBuffer(100, o -> logDropped(o), BufferOverflowStrategy.DROP_OLDEST)` will drop the oldest elements in a buffer of 100 elements that couldn't be consumed, and pass these dropped elements to the `logDropped` method.

Reply:
Interesting. This seems like something we could consider making configurable in the future. Right now, we're calling `onBackPressureLatest` on the `BufferingFlux`, which it seems we could add a consumer to while otherwise keeping the same with `onBackpressureBuffer(1, o -> handleDropped(o), BufferOverflowStrategy.DROP_OLDEST`